modules/tv-recommendations/android/src/main/java/expo/modules/tvrecommendations/TvRecommendationsPublisher.kt

```
@@ -96,126 +126,262 @@ internal object TvRecommendationsPublisher {
+    try {
+      val programIds = JSONObject(programIdsJson)
+      val keys = programIds.keys()
+      while (keys.hasNext()) {
+        val key = keys.next()
+        if (programIds.optLong(key, -1L) == programId) {
+          programIds.remove(key)
+          break
+        }
+      }
+      prefs.edit().putString(KEY_PROGRAM_IDS, programIds.toString()).apply()
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)